packages/api/src/controllers/taskController.js

```diff
@@ -237,22 +237,41 @@ const taskController = {
         if (assignee_user_id !== user_id) {
           return res.status(403).send('Not authorized to complete other people\'s task');
         }
-        const result = await TaskModel.transaction(async trx =>
-          await TaskModel.query(trx).context({ user_id: req.user.user_id })
-            .upsertGraph({ task_id: parseInt(task_id), ...data }, {
-              noUpdate: nonModifiable,
-              noDelete: true,
-              noInsert: true,
-            }),
+        const result = await TaskModel.transaction(async trx => {
```

Comment:
would the code be cleaner if you initialized a trx, followed by a `.commit` or `.rollback` depending on the result? i reckon that would be less nesting if that was the case.